docs/source/tools/trigger-service.rst

```diff
@@ -0,0 +1,119 @@
+
+Complex workflows can require running many triggers for many parties and at a certain point, use of ``daml trigger`` with its process per trigger model becomes unwieldy. The trigger-service provides the means to host multiple triggers for multiple parties running against a common ledger in a single process and provides a convenient interface for starting, stopping and monitoring them.
+
+The trigger-service is a ledger client that acts as an end-user agent. The trigger service intermediates between the ledger and end-users by running triggers on their behalf. The trigger-service is a REST service that interacts with end-users by exchanging HTTP requests and responses. Request/response data exchanged between the end-users and the trigger-service is in the JSON data-interchange format.
+
+Starting the trigger-service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, it is assumed there is a sandbox ledger running on port 6865 on the local host.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key
+
+The above starts the trigger-service using a number of default parameters. Most notably, the HTTP port the trigger service listens on which defaults to 8088. The meanining of the ``--no-secret-key`` parameter will be explained in the next section. To see all of the available parameters, their defaults and descriptions, one can execute the command ``daml trigger-service --help``.
+
+Although as we'll see, the trigger-service exposes an endpoint for end-users to upload DAR files to the service (DAML archives containing triggers), it is sometimes convenient to start the service preconfigured with a specific DAR. To do this, the ``--dar`` option is provided.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key --dar <DAR>
+
+Identifying trigger parties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we briefly explain how parties are identified in end-user interactions with a running trigger service.
+
+When an end-user interacts with a trigger-service, they do so by sending HTTP requests to one of several trigger-service HTTP endpoints. Such requests must contain information detailing the party the trigger-service is to act on the behalf of in its handling of the user request.
+
+This party information is conveyed via  HTTP basic authentication headers where the requesting party is represented by a unique username/password pair (Later versions of this documentation will explain running the trigger-service in authenticated contexts. For now, we assume non-authenticated contexts. In these contexts, the passwords associated with parties are not important and may have any values at all but they still must be provided).
```

Comment:
I wouldn’t be so detailed when it comes to later versions. Just state something like “for now the password is ignored. future versions of the trigger service will support authentication and validate passwords.”

Reply:
Ok!